src/compositor.rs

```diff
@@ -185,7 +185,7 @@ pub struct Surface(wl_surface::WlSurface);
 
 impl Surface {
     pub fn new<D>(
-        compositor: &impl ProvidesBoundGlobal<wl_compositor::WlCompositor, 5>,
+        compositor: &impl ProvidesBoundGlobal<wl_compositor::WlCompositor, 6>,
```

Comment:
Could you refactor the `6` into the version constant and use the constant in all places we use the version.